odl/solvers/functional/default_functionals.py

```diff
@@ -1655,7 +1707,8 @@ def __init__(self, *functionals):
         domain = ProductSpace(*domains)
         linear = all(func.is_linear for func in functionals)
 
-        super(SeparableSum, self).__init__(space=domain, linear=linear)
+        super(SeparableSum, self).__init__(domain=domain, linear=linear,
+                                           range=functionals[0].range)
```

Comment:
I'm not sure about this choice. One would perhaps expect that real+complex should be complex, but with this implementation it depends on the order.

Reply:
I understand your reasoning and personally I would agree with you. I just thought I went with the convention of operators, where the domain (including dtype) has to be the same. But you are certainly right, that this should not depend on the order of the ops. I will think about it